### PR TITLE
Allow to pass a commit SHA instead using GITHUB_SHA.

### PR DESCRIPTION
**Pull request by ysavary**
In some cases, when a previous step merged a branch into `github.ref` for example, I would like to specify the commit SHA to tag.

### CODE REVIEW

File: src/action.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export default async function main() {
   const dryRun = core.getInput('dry_run');
   const customReleaseRules = core.getInput('custom_release_rules');
   const shouldFetchAllTags = core.getInput('fetch_all_tags');
+  const commit_sha = core.getInput('commit_sha');
```
Comment (mathieudutour) [inline src/action.ts:29]: could you use camelCase to match the style of the rest of the code?

File: README.md [hunk 2]
```diff
@@ -31,6 +31,7 @@ jobs:
 ### 📥 Inputs
 
 - **github_token** _(required)_ - Required for permission to tag the repo. Usually `${{ secrets.GITHUB_TOKEN }}`.
+- **commit_sha** _(optional)_ - Use this commit SHA instead GITHUB_SHA.
```
Comment (mathieudutour) [inline README.md:34]: could you add a bit more context as to when someone would want to do this?

**Review by mathieudutour — CHANGES_REQUESTED** [review]

File: src/action.ts (hunk 1 above)
Comment (ysavary) [reply src/action.ts]: Oh yes, sure, sorry!

**Review by mathieudutour — APPROVED** [review]
Thanks!

**Pull request merged**